plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -461,6 +472,12 @@ public HiveMetadata(
         this.allowTableRename = allowTableRename;
         this.maxPartitionDropsPerQuery = maxPartitionDropsPerQuery;
         this.hiveViewsTimestampPrecision = requireNonNull(hiveViewsTimestampPrecision, "hiveViewsTimestampPrecision is null");
+        if (metadataParallelism == 1) {
+            this.metadataFetchingExecutor = directExecutor();
+        }
+        else {
+            this.metadataFetchingExecutor = newFixedThreadPool(metadataParallelism, daemonThreadsNamed("hive-metadata-%s"));
+        }
```

Comment:
This should use a BoundedExecutor over the common exectorService similar to the code for `max-concurrent-metastore-updates`